py-polars/tests/unit/operations/test_group_by.py

```diff
@@ -495,202 +495,69 @@ def test_group_by_empty_groups(
-            "27",
-            "43",
-            "45",
-            "41",
-            "13",
-            "45",
-            "48",
-            "17",
-            "22",
-            "31",
-            "25",
-            "28",
-            "13",
-            "7",
-            "26",
-            "17",
-            "4",
-            "43",
-            "47",
-            "30",
-            "28",
-            "8",
-            "27",
-            "6",
-            "7",
-            "26",
-            "11",
-            "37",
-            "29",
-            "49",
-            "20",
-            "29",
-            "28",
-            "23",
-            "9",
-            None,
-            "38",
-            "19",
-            "7",
-            "38",
-            "3",
-            "30",
-            "37",
-            "41",
-            "5",
-            "16",
-            "26",
-            "31",
-            "6",
-            "25",
-            "11",
-            "17",
-            "31",
-            "31",
-            "20",
-            "26",
-            None,
-            "39",
-            "10",
-            "38",
-            "4",
-            "39",
-            "15",
-            "13",
-            "35",
-            "38",
-            "11",
-            "39",
-            "11",
-            "48",
-            "36",
-            "18",
-            "11",
-            "34",
-            "16",
-            "28",
-            "9",
-            "37",
-            "8",
-            "17",
-            "48",
-            "44",
-            "28",
-            "25",
-            "30",
-            "37",
-            "30",
-            "18",
-            "12",
-            None,
-            "27",
-            "10",
-            "3",
-            "16",
-            "27",
-            "6",
-        ],
-        dtype=pl.Categorical,
+# https://github.com/pola-rs/polars/issues/8663
+def test_perfect_hash_table_null_values() -> None:
+    # fmt: off
+    values = ["3", "41", "17", "5", "26", "27", "43", "45", "41", "13", "45", "48", "17", "22", "31", "25", "28", "13", "7", "26", "17", "4", "43", "47", "30", "28", "8", "27", "6", "7", "26", "11", "37", "29", "49", "20", "29", "28", "23", "9", None, "38", "19", "7", "38", "3", "30", "37", "41", "5", "16", "26", "31", "6", "25", "11", "17", "31", "31", "20", "26", None, "39", "10", "38", "4", "39", "15", "13", "35", "38", "11", "39", "11", "48", "36", "18", "11", "34", "16", "28", "9", "37", "8", "17", "48", "44", "28", "25", "30", "37", "30", "18", "12", None, "27", "10", "3", "16", "27", "6"]
+    groups = ["3", "41", "17", "5", "26", "27", "43", "45", "13", "48", "22", "31", "25", "28", "7", "4", "47", "30", "8", "6", "11", "37", "29", "49", "20", "23", "9", None, "38", "19", "16", "39", "10", "15", "35", "36", "18", "34", "44", "12"]
+    # fmt: on
+
+    s = pl.Series("a", values, dtype=pl.Categorical)
+
+    result = (
+        s.to_frame("a").group_by("a", maintain_order=True).agg(pl.col("a").alias("agg"))
```

Comment:
`maintain_order=True` is now required because it can no longer take the fast path where the order is guaranteed.